code/modules/mob/living/basic/jungle/leaper/leaper_abilities.dm

```diff
@@ -0,0 +1,255 @@
+
+/obj/effect/temp_visual/blood_drop_rising/Initialize(mapload)
+	. = ..()
+	animate(src, pixel_y = base_pixel_y + 150, time = duration)
+
+/obj/effect/temp_visual/blood_drop_falling
+	name = "leaper bubble"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "blood_effect_falling"
+	layer = ABOVE_ALL_MOB_LAYER
+	plane = GAME_PLANE_UPPER_FOV_HIDDEN
+	duration = 0.7 SECONDS
+	pixel_y = 60
+
+/obj/effect/temp_visual/blood_drop_falling/Initialize(mapload)
+	. = ..()
+	addtimer(CALLBACK(src, PROC_REF(create_blood_structure)), duration)
+	animate(src, pixel_y = 0, time = duration)
+
+/obj/effect/temp_visual/blood_drop_falling/proc/create_blood_structure()
+	playsound(src, 'sound/effects/snap.ogg', 50, TRUE)
+	new /obj/structure/leaper_bubble(get_turf(src))
+
+/obj/effect/temp_visual/shadow_telegraph
+	name = "shadow"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "shadow_telegraph"
+	duration = 1.5 SECONDS
+
+/obj/effect/decal/cleanable/leaper_sludge
+	name = "leaper sludge"
+	desc = "A small pool of sludge, containing trace amounts of leaper venom."
+	icon = 'icons/effects/tomatodecal.dmi'
+	icon_state = "tomato_floor1"
+
+/obj/structure/leaper_bubble
+	name = "leaper bubble"
+	desc = "A floating bubble containing leaper venom. The contents are under a surprising amount of pressure."
+	icon = 'icons/obj/weapons/guns/projectiles.dmi'
+	icon_state = "leaper"
+	max_integrity = 10
+	density = FALSE
+
+/obj/structure/leaper_bubble/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/movetype_handler)
+	ADD_TRAIT(src, TRAIT_MOVE_FLOATING, LEAPER_BUBBLE_TRAIT)
+	QDEL_IN(src, 100)
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_entered),
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+
+/obj/structure/leaper_bubble/Destroy()
+	new /obj/effect/temp_visual/leaper_projectile_impact(get_turf(src))
+	playsound(src,'sound/effects/snap.ogg', 50, TRUE)
+	return ..()
+
+/obj/structure/leaper_bubble/proc/on_entered(datum/source, atom/movable/bubbled)
+	SIGNAL_HANDLER
+	if(!isliving(bubbled) || istype(bubbled, /mob/living/basic/leaper))
+		return
+	var/mob/living/bubbled_mob = bubbled
+
+	playsound(src, 'sound/effects/snap.ogg',50, TRUE)
+	bubbled_mob.Paralyze(50)
+	if(iscarbon(bubbled_mob))
+		bubbled_mob.reagents.add_reagent(/datum/reagent/toxin/leaper_venom, 5)
+	else if(isanimal(bubbled_mob))
+		var/mob/living/simple_animal/bubbled_animal = bubbled_mob
+		bubbled_animal.adjustHealth(25)
+	else if(isbasicmob(bubbled_mob))
+		bubbled_mob.adjustBruteLoss(25)
+	qdel(src)
+
+/datum/reagent/toxin/leaper_venom
+	name = "Leaper venom"
+	description = "A toxin spat out by leapers that, while harmless in small doses, quickly creates a toxic reaction if too much is in the body."
+	color = "#801E28" // rgb: 128, 30, 40
+	toxpwr = 0
+	taste_description = "french cuisine"
+	taste_mult = 1.3
+
+/datum/reagent/toxin/leaper_venom/on_mob_life(mob/living/carbon/M, seconds_per_tick, times_fired)
+	. = ..()
+	if(volume < 10)
```

Comment:
Does it not do anything to a carbon unless they get bubbled twice then? Or am I missing something?